llvm/lib/Transforms/IPO/FunctionAttrs.cpp

```diff
@@ -118,7 +118,7 @@ static void addLocAccess(MemoryEffects &ME, const MemoryLocation &Loc,
   if (isNoModRef(MR))
     return;
 
-  const Value *UO = getUnderlyingObject(Loc.Ptr);
+  const Value *UO = getUnderlyingObjectAggressive(Loc.Ptr);
   assert(!isa<AllocaInst>(UO) &&
          "Should have been handled by getModRefInfoMask()");
```

Comment:
Should convert this into `if(isa<AllocaInst>(UO)) return`.

Reply:
Thanks! I see it in the logs now. It doesn't seem to require additional test cases, so I will submit it directly later.

Reply:
Reland with a test case: https://github.com/llvm/llvm-project/commit/559be8e2b55afbf6736fed646542fb25793f0f7e.